nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/credentials/provider/factory/strategies/WebIdentityCredentialsStrategy.java

```diff
@@ -0,0 +1,219 @@
+                                                         final String roleArn,
+                                                         final String roleSessionName,
+                                                         final Integer sessionSeconds) {
+            this.stsClient = Objects.requireNonNull(stsClient, "stsClient required");
+            this.oauth2AccessTokenProvider = Objects.requireNonNull(oauth2AccessTokenProvider, "OAuth2AccessTokenProvider required");
+            this.roleArn = Objects.requireNonNull(roleArn, "roleArn required");
+            this.roleSessionName = Objects.requireNonNull(roleSessionName, "roleSessionName required");
+            this.sessionSeconds = sessionSeconds;
+        }
+
+        @Override
+        public AwsCredentials resolveCredentials() {
+            final Instant now = Instant.now();
+            final AwsSessionCredentials current = cached;
+            final Instant currentExpiration = expiration;
+            if (current != null && currentExpiration != null && now.isBefore(currentExpiration.minus(SKEW))) {
+                return current;
+            }
+
+            synchronized (this) {
+                if (cached != null && expiration != null && Instant.now().isBefore(expiration.minus(SKEW))) {
+                    return cached;
+                }
+
+                final String webIdentityJwt = getWebIdentityToken();
+
+                final AssumeRoleWithWebIdentityRequest.Builder reqBuilder = AssumeRoleWithWebIdentityRequest.builder()
+                        .roleArn(roleArn)
+                        .roleSessionName(roleSessionName)
+                        .webIdentityToken(webIdentityJwt);
+
+                if (sessionSeconds != null) {
+                    reqBuilder.durationSeconds(sessionSeconds);
+                }
+
+                final AssumeRoleWithWebIdentityResponse resp = stsClient.assumeRoleWithWebIdentity(reqBuilder.build());
+                final Credentials creds = resp.credentials();
+                final AwsSessionCredentials sessionCreds = AwsSessionCredentials.create(
+                        creds.accessKeyId(), creds.secretAccessKey(), creds.sessionToken());
+
+                this.cached = sessionCreds;
+                this.expiration = creds.expiration();
+                return sessionCreds;
+            }
+        }
+
+        private String getWebIdentityToken() {
+            final AccessToken accessToken = oauth2AccessTokenProvider.getAccessDetails();
+            if (accessToken == null) {
+                throw new IllegalStateException("OAuth2AccessTokenProvider returned null AccessToken");
+            }
+
+            // Prefer id_token when available
+            final Map<String, Object> additional = accessToken.getAdditionalParameters();
+            if (additional != null) {
+                final Object idTokenObj = additional.get("id_token");
+                if (idTokenObj instanceof String && !((String) idTokenObj).isEmpty()) {
```

Comment:
For simplicity, and clarity of behavior, I recommend simply checking of `idToken` is not null, and the throwing an `IllegalStateException` if the string itself is empty.